app/src/main/java/net/squanchy/tweets/service/TwitterService.java

```diff
@@ -0,0 +1,129 @@
+package net.squanchy.tweets.service;
+
+import com.twitter.sdk.android.core.Callback;
+import com.twitter.sdk.android.core.Result;
+import com.twitter.sdk.android.core.TwitterException;
+import com.twitter.sdk.android.core.models.Identifiable;
+import com.twitter.sdk.android.tweetui.Timeline;
+import com.twitter.sdk.android.tweetui.TimelineResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.squanchy.tweets.view.TimelineStateHolder;
+
+public class TwitterService<T extends Identifiable> {
```

Comment:
We can actually remove the generic and just go with `Tweet` as we won't be using it with anything else